woocommerce/woocommerce-gutenberg-products-block
# Update ProductsControl to use API search when product catalog is large

Fixes #505 – If a store has more than 200 products, the existing products search list can take 10+ seconds to load, which negates the performance benefit of the existing search. To balance this, we can switch to using pagination and an API search process if the store has 200+ products.

### How to test the changes in this Pull Request:

~Since the `onSearch` prop is not added to the SearchListControl yet, you can test this and see there is no functionality change, or you can tell the component to use the local file, with [this branch checked out.](https://github.com/woocommerce/woocommerce-admin/pull/1992)~ The `@woocommerce/components` package has been published, you can test this properly now. Make sure to run `npm install` before building.

To test, add a Hand-picked Products block (the product list in Featured Products will also get this treatment, if this looks good, but it's not yet). Before you add the block, open the Network tab and watch the XHR requests.

If your store has less than 200 products, you'll only see the API calls when the ProductsControl loads. Searching the list should be quick, and not trigger any other API requests.

If your store has over 200 products, you'll see two API calls when the ProductsControl loads, one for the first page of products, and just one for the selected products. Each search will also trigger two API requests, for your search term, and one for the selected products again, since we need to keep them in the list.

In any case, you should be able to find the products you're searching for.
